lib/gh_importer/mapping/dmca.rb

```diff
@@ -0,0 +1,170 @@
+require 'gh_importer/entity_notice_role_builder'
+
+module GithubImporter
+  module Mapping
+    class DMCA
+      NOTICE_TYPE_LABEL = 'DMCA'
+      GITHUB = 'Github'
+      URLS_TO_IGNORE = ['https://github.com/github/dmca/blob/master/README.md#anatomy-of-a-takedown-notice).', 'https://docs.github.com/en/github/site-policy/dmca-takedown-policy#a-how-does-this-actually-work).', 'https://docs.github.com/en/articles/guide-to-submitting-a-dmca-counter-notice).']
+
+      def initialize(notice_text, filename)
+        @notice_text = notice_text
+        @notice_type = DMCA
+        @filename = filename
+      end
+
+      def notice_type
+        ::DMCA
+      end
+
+      def title
+        "Takedown Request regarding #{self.class::NOTICE_TYPE_LABEL} Complaint to Github"
+      end
+
+      def entity_notice_roles
+        [
+          github_entity_role('submitter'),
+          github_entity_role('recipient'),
+          principal
+        ]
+      end
+
+      def file_uploads
+        []
+      end
+
+      def mark_registration_number
+        ''
+      end
+
+      def works
+        works = parse_works
+        return works if works.any?
+
+        []
+      end
+
+      def subject
+        'DMCA notice to GitHub'
+      end
+
+      def topics
+        Topic.where(name: self.class::NOTICE_TYPE_LABEL)
+      end
+
+      def source
+        'GitHub DMCA repository'
+      end
+
+      def tag_list
+        ''
+      end
+
+      def action_taken
+        ''
+      end
+
+      def body
+        @notice_text
+      end
+
+      def body_original
+        ''
+      end
+
+      def jurisdiction
+        ['US']
+      end
+
+      def regulation_list
+        []
+      end
+
+      def language
+        return 'en'
+      end
+
+      def local_jurisdiction_laws
+        ''
+      end
+
+      private
+
+      def parse_mark_registration_number(*)
+      end
+
+      def sender(*)
+      end
+
+      def sender_address(*)
+      end
+
+      def principal(*)
+        name = @filename.match(/\d{4}-\d{2}-\d{2}-(.*?)\.md/m).captures
+        titleized_name = name[0].tr('-', ' ').titleize
+        build_role('principal', titleized_name)
+      end
+
+      def recipient(*)
+      end
+
+      def attorney(*)
+      end
+
+      def github_entity_role(role_name)
+        existing_entity = EntityNoticeRole.new(
```

Comment:
Here it will be safer to find it by `User.where(email: 'gh@email.com').entities.first`. It would be cool to have the email address configurable. You can create a setting using a migration like here: https://github.com/berkmancenter/lumendatabase/blob/42f4db696aaf851ce076e9a4eecd135556e213fd/db/migrate/20210917184944_create_setting_for_scale_of_mention.rb.

Then you can read it using `LumenSetting.get('github_user_email')`.